core/optaplanner-core-impl/src/main/java/org/optaplanner/core/impl/domain/common/accessor/CachedMemberAccessorFactory.java

```diff
@@ -0,0 +1,45 @@
+package org.optaplanner.core.impl.domain.common.accessor;
+
+import java.lang.annotation.Annotation;
+import java.lang.reflect.Member;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+import org.optaplanner.core.api.domain.common.DomainAccessType;
+import org.optaplanner.core.impl.domain.common.accessor.gizmo.GizmoMemberAccessorFactory;
+
+public final class CachedMemberAccessorFactory {
```

Comment:
Not sure why the word `Cached` is in there. Let's make it shorter, omit it? It's an implementation detail.

Reply:
The `MemberAccessorFactory` would have two `buildMemberAccessor()` methods, one of them static, with the exact same parameters. The implementation detail will leak anyway in one of the methods' names.